### DISCUSSION TITLE
Un juego para aprender a hacer prompt injection

### DISCUSSION THREAD

**Discussion by harpomaxx** [harpomaxx/algoritmos2#25] [General]
Con la gente de la [CTU](https://www.stratosphereips.org/) (czech Techincal University) armamos este jueguito para aprender sobre [prompt injection](https://learnprompting.org/docs/prompt_hacking/injection#:~:text=Prompt%20injection%20is%20the%20process,as%20part%20of%20the%20prompt.) a los [LLMs](https://en.wikipedia.org/wiki/Language_model) (como ChatGPT). 

La idea es tratar de lograr que chatGPT diga una palabra secreta. En cada nivel se le instruyo a ChatGPT que actue de cierta manera para que no entregue el password. Sin embargo es posible convencerlo... si se sabe como.

idea para aprender como funcionan los LLMs. Sus alcances, limitaciones y sobre todo como hacer prompt injection ;-)
 
 http://147.32.83.132:8000/
 
 
![image](https://github.com/harpomaxx/algoritmos2/assets/12446313/2a0eb9fa-daea-45e0-b090-fdc5b85085f1)

**Comment by franualba** [comment]
En el nivel "Noob" me salió al 4to intento, y me quedó medio largo el párrafo. No me imagino lo que habrá que hacer en el nivel "God"... jaja

![image](https://github.com/harpomaxx/algoritmos2/assets/127322698/7b1cfc41-d00c-4c89-9db9-8312ebc71729)

**Reply by harpomaxx** [reply]
soporta spanish ojo, en el fondo esta chatGPT asi que...

**Reply by Rolomza** [reply]
Está muy entretenido profe, gracias por compartirlo...

<img width="1159" alt="image" src="https://github.com/harpomaxx/algoritmos2/assets/56059292/a380be84-0726-4dd6-97ed-3794b6420cfb">

**Reply by franualba** [reply]
> Está muy entretenido profe, gracias por compartirlo...
> <img alt="image" width="1159" src="https://user-images.githubusercontent.com/56059292/238521702-a380be84-0726-4dd6-97ed-3794b6420cfb.png">

No entiendo muy bien qué estaría pasando. Le pasé la misma fecha que Augusto por ejemplo para ver cómo era posible que te mostrara un "historial" si supuestamente cada prompt es independiente del anterior, y resulta que me largó una supuesta conversación con prompts que nunca le dí:

![image](https://github.com/harpomaxx/algoritmos2/assets/127322698/9e783d94-9074-46a0-ad91-4546182ee10a)

Puede ser un bug o es normal?

**Reply by harpomaxx** [reply]
Hola, recordemos que los LLMs son no-deterministico. Es decir frente a un mismo prompt puede no necesariamente responder de la misma manera. Tienen un parámetro `temperature` que de alguna manera influye en la distribución de probabilidad y lo fuerza a ser más "creativo". En este caso, ese parámetro fue asignado a un valor muy alto. 

Respecto a la respuesta... es el resultado de ese cálculo probabilístico. No es un bug. Frente a esta entrada... GPT ha elegido las palabras más probables... y bueno, terminó generando esta conversación que obviamente nunca existió

**Reply by franualba** [reply]
Qué interesante, y qué loco también.... Me agarró ansiedad de cursar Inteligencia Artifical ahora, jaja

**Reply by harpomaxx** [reply]
Es importante conocer como funcionan, por que hay una locura "importante" en los medios prediciendo el reemplazo de todos nosotros por estos algoritmos... que como ven.. si bien son muy potentes.. no dejan de ser herramientas para ayudarnos en algunas tareas... en otras todavia no ;-)

**Comment by PaulaMartinez48** [comment]
Está bueno profe. Estuve un buen rato tratando de convencer a la IA que le podía dar libertad y me la terminó dando jajasj
![Sin título](https://github.com/harpomaxx/algoritmos2/assets/84521246/e8c98ad1-1f28-42c2-9497-65a66a7905cc)

**Comment by harpomaxx** [comment]
Gracias por probarlo!!! :hugs:
